driver/src/test/scala/org/mongodb/scala/internal/ScalaObservableSpec.scala

```diff
@@ -406,6 +406,11 @@ class ScalaObservableSpec extends FlatSpec with Matchers {
     Option(Await.result(TestObservable[Int](Observable(List[Int]())).head(), Duration(10, TimeUnit.SECONDS))) should equal(None)
   }
 
+  it should "not stackoverflow when using flatMap with execution contexts" in {
```

Comment:
Can you explain what's causing the stack overflow with the current implementation?

Reply:
The stack with this:

```
observable(1 to 10000).flatMap((res: Int) => Observable(Seq(res)))
```
Essentially creates 10000 observables, one for each of the items in the original observable.